utils/collections-management.in

```diff
@@ -698,17 +698,41 @@ print_list_of_items () {
   printf "\n"
 }
 
-
 get_password () {
-  printf "Enter Password : "
-  stty -echo
-  trap 'stty echo' EXIT
-  read PASSWORD
-  stty echo
-  trap - EXIT
-  printf "\n"
-  if ! (printf -- "%s\n" "${PASSWORD}" | \
-          sudo -k -S printf "" > /dev/null 2>&1); then
+  # To be sure of reading data from the terminal no matter
```

Comment:
I guess this would exclude the possibility of using the script in an automated way?  Is that correct?

Reply:
@ellio167 I am not sure about this. I used a similar approach before in my scripts since especially for passwords with white spaces at the start. But I do not know about any limitation for using the script in an automated way. We have to check.

Reply:
It seems that using /dev/tty is the standard practice of sudo and other programs. Indeed, here I use the `-S` argument to sudo in order to take the password from stdin....

(I found this useful: https://superuser.com/questions/67765/sudo-with-password-in-one-command-line)

So, after reminding myself of these details, I'm very happy with this as is.

Reply:
@ellio167 I agree with the `-S` part, but the major problem is on `read PASSWORD` if the entered password starts with e.g. white space or some other special characters the reader will trim them, and it would fail with the Bad password message since it git the wrong password.
Please try it with different passwords and you would see. I encountered this problem many times before and I know many people have it.

Reply:
@yafshar Sorry, I think my message was confusing.  My "as is" was meant to refer to your improved code.  I like what you have provided.

I have a couple of other small edits that I'll push here in a minute.